src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java

```diff
@@ -411,16 +411,30 @@ public static void setLocale(Locale locale) {
 
     @Override
     public int compareTo(LocalizedMessage other) {
-        int result = Integer.compare(lineNo, other.lineNo);
+        final int result;
 
         if (lineNo == other.lineNo) {
             if (columnNo == other.columnNo) {
-                result = getMessage().compareTo(other.getMessage());
+                if (Objects.equals(moduleId, other.moduleId)) {
+                    result = getMessage().compareTo(other.getMessage());
```

Comment:
@romani I don't see us comparing non-id sources of message.
If 2 different modules print the exact same message on the same line/column, we will have the same issue as before.
Should this be a new issue where we add module name if `id` is not specified to the compare?